payment-processor/src/main/kotlin/org/ostelco/prime/paymentprocessor/StripePaymentProcessor.kt

```diff
@@ -204,52 +204,61 @@ class StripePaymentProcessor : PaymentProcessor {
 
     override fun authorizeCharge(customerId: String, sourceId: String?, amount: Int, currency: String): Either<PaymentError, String> {
         val errorMessage = "Failed to authorize the charge for customerId $customerId sourceId $sourceId amount $amount currency $currency"
-        return either(errorMessage) {
-            val chargeParams = mutableMapOf(
-                    "amount" to amount,
-                    "currency" to currency,
-                    "customer" to customerId,
-                    "capture" to false)
-            if (sourceId != null) {
-                chargeParams["source"] = sourceId
+        return when {
+            amount == 0 -> Either.right("ZERO_CHARGE_IS_OK")
+            else -> either(errorMessage) {
+                val chargeParams = mutableMapOf(
+                        "amount" to amount,
+                        "currency" to currency,
+                        "customer" to customerId,
+                        "capture" to false)
+                if (sourceId != null) {
+                    chargeParams["source"] = sourceId
+                }
+                Charge.create(chargeParams)
+            }.flatMap { charge: Charge ->
+                val review = charge.review
+                Either.cond(
+                        test = (review == null),
+                        ifTrue = { charge.id },
+                        ifFalse = { ForbiddenError("Review required, $errorMessage $review") }
+                )
             }
-            Charge.create(chargeParams)
-        }.flatMap { charge: Charge ->
-            val review = charge.review
-            Either.cond(
-                    test = (review == null),
-                    ifTrue = { charge.id },
-                    ifFalse = { ForbiddenError("Review required, $errorMessage $review") }
-            )
         }
     }
 
     override fun captureCharge(chargeId: String, customerId: String): Either<PaymentError, String> {
         val errorMessage = "Failed to capture charge for customerId $customerId chargeId $chargeId"
-        return either(errorMessage) {
-            Charge.retrieve(chargeId)
-        }.flatMap { charge: Charge ->
-            val review = charge.review
-            Either.cond(
-                    test = (review == null),
-                    ifTrue = { charge },
-                    ifFalse = { ForbiddenError("Review required, $errorMessage $review") }
-            )
-        }.flatMap { charge ->
-            try {
-                charge.capture()
-                Either.right(charge.id)
-            } catch (e: Exception) {
-                logger.warn(errorMessage, e)
-                Either.left(BadGatewayError(errorMessage))
+        return when {
```

Comment:
I think we should pass the amount parameter and check that value instead of checking the chargeId